### ISSUE TITLE
CSP: FileUpload Missing Nonce

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#5641]
```xml
<h:form id="frmUpload">
   <p:fileUpload 
		listener="#{imageController.upload}"
		mode="advanced" showButtons="false" dragDropSupport="true"
		auto="true" allowTypes="/(\.|\/)(csv)$/"
		label="#{webmsg['action.file.choose']}"
		style="padding:6px;margin-bottom:6px;word-break:break-all;"
		update="@form"/>
</h:form>
```

After running the upload with CSP enabled we get this stack trace:

```
SEVERE [javax.enterprise.resource.webcontainer.jsf.application] (default task-8) Error Rendering View[/pages/public/public-dashboard.xhtml]: javax.faces.FacesException: Missing CSP nonce
       at org.primefaces.csp.CspState.validate(CspState.java:75)
       at org.primefaces.csp.CspState.getNonce(CspState.java:57)
       at org.primefaces.csp.CspResponseWriter.listenOnEndAttribute(CspResponseWriter.java:185)
       at org.primefaces.csp.CspResponseWriter.write(CspResponseWriter.java:167)
       at org.primefaces.csp.CspPartialResponseWriter.write(CspPartialResponseWriter.java:167)
       at javax.faces.context.ResponseWriterWrapper.write(ResponseWriterWrapper.java:380)
       at java.io.Writer.write(Writer.java:192)
       at java.io.Writer.write(Writer.java:157)
       at org.primefaces.util.WidgetBuilder.init(WidgetBuilder.java:59)
       at org.primefaces.util.WidgetBuilder.init(WidgetBuilder.java:78)
       at org.primefaces.component.fileupload.FileUploadRenderer.encodeScript(FileUploadRenderer.java:93)
       at org.primefaces.component.fileupload.FileUploadRenderer.encodeEnd(FileUploadRenderer.java:83)
       at javax.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:920)
       at javax.faces.component.UIComponent.encodeAll(UIComponent.java:1863)
       at javax.faces.render.Renderer.encodeChildren(Renderer.java:176)
```

**Comment by melloware** [comment]
Still trying to reproduce this standalone.

[pf-5641.zip](https://github.com/primefaces/primefaces/files/4294764/pf-5641.zip)

**Comment by melloware** [comment]
Closing this was fixed between RC3 and 8.0 Final.
